src/LiquidityPool.sol

```diff
@@ -270,46 +253,40 @@ contract LiquidityPool is Auth {
         return share.totalSupply();
     }
 
-    function balanceOf(address owner) public returns (uint256) {
+    function balanceOf(address owner) public view returns (uint256) {
         return share.balanceOf(owner);
     }
 
-    function transferFrom(address sender, address recipient, uint256 amount)
-        public
-        withTokenApproval(sender, amount)
-        returns (bool)
-    {
-        // discuss if we should add this here
-        // approveForOwner(sender, address(this), amount);
-        return share.transferFrom(sender, recipient, amount);
-    }
-
-    function transfer(address recipient, uint256 amount) public returns (bool) {
-        return transferFrom(msg.sender, recipient, amount);
+    function allowance(address owner, address spender) public view returns (uint256) {
```

Comment:
We have the generic ERC20 tests in `test/ERC20.t.sol`, we should adapt these to also be used for the liquidity pool to ensure it is compliant.